server/src/main/java/org/apache/kafka/server/ClientMetricsManager.java

```diff
@@ -112,7 +113,7 @@ public ClientMetricsManager(ClientMetricsReceiverPlugin receiverPlugin, int clie
         this.subscriptionMap = new ConcurrentHashMap<>();
         this.subscriptionUpdateVersion = new AtomicInteger(0);
         this.clientInstanceCache = new SynchronizedCache<>(new LRUCache<>(CACHE_MAX_SIZE));
-        this.expirationTimer = new SystemTimerReaper("client-metrics-reaper", new SystemTimer("client-metrics"));
+        this.expirationTimer = new SystemTimerReaper(CLIENT_METRICS_REAPER_THREAD_NAME, new SystemTimer("client-metrics"));
```

Comment:
Could you add test to check  `SystemTimerReaper#close` does close the system timer? The system timer has inner thread also.

Reply:
Already add a test in 5f220e297990311d631544582ac27d11473ebea4 while the test need some additional methods to check if executorservice/thread is shutdown or not. Or maybe we can simply add `executor-` (the thread prefix used by SystemTimer ) to `unexpectedThreadNames`. Let the unexpectedThreadNames to check if thread are cleanup or not.
https://github.com/apache/kafka/blob/dc189a39cc79d45c59fa1762ca9379bdbd85ee34/server-common/src/main/java/org/apache/kafka/server/util/timer/SystemTimer.java#L52

Reply:
I just notice this response. Could you add `executor-`  to `unexpectedThreadNames` to trigger QA again?